core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -0,0 +1,309 @@
+         * */
+        private final List<Consumer<? super Server>> serverStoppingCallbacks;
+
+        /**
+         * {@link Consumer}s invoked when the {@link Server} is stopped.
+         * */
+        private final List<Consumer<? super Server>> serverStoppedCallbacks;
+
+        CallbackServerListener(List<Consumer<? super Server>> serverStartingCallbacks,
+                               List<Consumer<? super Server>> serverStartedCallbacks,
+                               List<Consumer<? super Server>> serverStoppingCallbacks,
+                               List<Consumer<? super Server>> serverStoppedCallbacks) {
+            this.serverStartingCallbacks = ImmutableList.copyOf(serverStartingCallbacks);
+            this.serverStartedCallbacks = ImmutableList.copyOf(serverStartedCallbacks);
+            this.serverStoppingCallbacks = ImmutableList.copyOf(serverStoppingCallbacks);
+            this.serverStoppedCallbacks = ImmutableList.copyOf(serverStoppedCallbacks);
+        }
+
+        @Override
+        public void serverStarting(Server server) {
+            for (Consumer<? super Server> serverStartingCallback : serverStartingCallbacks) {
+                serverStartingCallback.accept(server);
+            }
+        }
+
+        @Override
+        public void serverStarted(Server server) {
+            for (Consumer<? super Server> serverStartedCallback : serverStartedCallbacks) {
+                serverStartedCallback.accept(server);
+            }
+        }
+
+        @Override
+        public void serverStopping(Server server) {
+            for (Consumer<? super Server> serverStoppingCallback : serverStoppingCallbacks) {
+                serverStoppingCallback.accept(server);
+            }
+        }
+
+        @Override
+        public void serverStopped(Server server) {
+            for (Consumer<? super Server> serverStoppedCallback : serverStoppedCallbacks) {
+                serverStoppedCallback.accept(server);
+            }
+        }
+    }
+
+    /**
+     * Add {@link Runnable} invoked when the {@link Server} is starting.
+     * (see: {@link ServerListener#serverStarting(Server)})
+     */
+    public ServerListenerBuilder addStartingCallback(Runnable runnable) {
+        requireNonNull(runnable, "runnable");
+        serverStartingCallbacks.add((unused) -> runnable.run());
+        return this;
+    }
+
+    /**
+     * Add {@link Consumer} invoked when the {@link Server} is starting.
+     * (see: {@link ServerListener#serverStarting(Server)})
+     */
+    public ServerListenerBuilder addStartingCallback(Consumer<? super Server> consumer) {
+        serverStartingCallbacks.add(requireNonNull(consumer, "consumer"));
+        return this;
+    }
+
+    /**
+     * Add {@link Consumer}s invoked when the {@link Server} is starting.
+     * (see: {@link ServerListener#serverStarting(Server)})
+     */
+    public ServerListenerBuilder addStartingCallbacks(Consumer<? super Server>... consumers) {
```

Comment:
Delegate all array versions to iterable versions, e.g.,

`addStartingCallbacks(Arrays.asList(consumers))`

Reply:
Sure.